lib/MusicBrainz/Server/Controller/Role/Cleanup.pm

```diff
@@ -2,14 +2,30 @@ package MusicBrainz::Server::Controller::Role::Cleanup;
 use Moose::Role;
 use namespace::autoclean;
 
+use MusicBrainz::Server::Constants qw( entities_with );
+
+my %model_to_edit_type = entities_with('add_edit_type', take => sub {
+    my ($type, $properties) = @_;
+    return ($properties->{model} => $properties->{add_edit_type});
+});
+
 after show => sub {
     my ($self, $c) = @_;
     my $entity = $c->stash->{entity};
-    my $eligible_for_cleanup = $c->model( $self->config->{model} )->is_empty($entity->id);
+    my $model = $self->config->{model};
+    my $creation_edit = $c->model('Edit')->find_creation_edit(
+        $model_to_edit_type{$model},
+        $entity->id);
+    # If the creation edit is the only pending edit
+    # and the entity is otherwise empty,
+    # it will be removed as soon as the creation edit passes
+    my $in_cleanup_danger = $entity->edits_pending == 1 &&
+        $creation_edit && $creation_edit->is_open &&
+        $c->model($model)->is_empty($entity->id, ignore_edits => 1);
+    my $eligible_for_cleanup = $c->model($model)->is_empty($entity->id);
```

Comment:
Hmm. Do we need separate banners for these two cases? It seems like we could just remove all the edit-checking code from `is_empty` and check here if `edits_pending == 0 || (edits_pending == 1 && creation_edit->is_open)`, then just show a generic message about the entity being in danger of removal within the next few days.

(But we should also change the `RemoveEmpty` script so that that's actually true; it appears to remove things that are empty after only 1 day.)

Reply:
Yes, I think it's always been 1 day, "the next few days" was always in case it failed to run or something :smile: 

We'd still need to check if it's empty, right? It might be other edits adding more data have been made as autoedits. But that could check only for data and not edits, yes.

Reply:
Right, we'd still check if it's empty in any case. One banner for both cases + changing the RemoveEmpty interval to 3 days or something makes sense to me. Do you see any issue with combining them?

Reply:
The only issue with moving the RemoveEmpty interval to 3 days is that it adds more risk of someone actually using the stuff before it gets deleted (say, if an editor empties an incorrect label in hopes of it getting removed). Not sure how common that is, though.

Reply:
Another issue is for stuff like vandalism you'd probably want to get rid ASAP once empty, but I guess for that case we could add a manual way of doing it :)

Reply:
So if we did this, what would remain in `is_empty`? Something like this only?

```
    my $used_in_relationship = used_in_relationship($self->c, artist => 'artist_row.id');
    return $self->sql->select_single_value(<<~"SQL", $artist_id);
        SELECT TRUE
        FROM artist artist_row
        WHERE id = ?
        AND NOT (
          EXISTS (
            SELECT TRUE FROM artist_credit_name
            WHERE artist = artist_row.id
            LIMIT 1
          ) OR
          $used_in_relationship
        )
        SQL
```

I guess the `RemoveEmpty` script does its own separate check for emptiness so that wouldn't necessarily be a problem to remove from here.

Also, I'm assuming we'd have a separate ticket for the 1 day -> 3 days RemoveEmpty change? :)

Reply:
I tried to implement this more or less as talked, see if it looks good now :)